common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -769,58 +775,64 @@ class @Problem
     # params:
     #   'enable' is a boolean to determine enabling/disabling of buttons.
     #   'isFromCheckOperation' is a boolean to keep track if operation was initiated
-    #    from @check so that text of check button will also be changed while disabling/enabling
-    #    the check button.
+    #    from @submit so that text of submit button will also be changed while disabling/enabling
+    #    the submit button.
     if enable
       @resetButton
         .add(@saveButton)
         .add(@hintButton)
         .add(@showButton)
-        .removeClass('is-disabled')
         .removeAttr 'disabled'
     else
       @resetButton
         .add(@saveButton)
         .add(@hintButton)
         .add(@showButton)
-        .addClass('is-disabled')
         .attr({'disabled': 'disabled'})
 
-    @enableCheckButton enable, isFromCheckOperation
+    @enableSubmitButton enable, isFromCheckOperation
 
-  enableCheckButton: (enable, changeText = true) =>
-    # Used to disable check button to reduce chance of accidental double-submissions.
+  enableSubmitButton: (enable, changeText = true) =>
+    # Used to disable submit button to reduce chance of accidental double-submissions.
     # params:
-    #   'enable' is a boolean to determine enabling/disabling of check button.
+    #   'enable' is a boolean to determine enabling/disabling of submit button.
     #   'changeText' is a boolean to determine if there is need to change the
-    #    text of check button as well.
-    if enable
-      @checkButton.removeClass 'is-disabled'
-      @checkButton.removeAttr 'disabled'
+    #    text of submit button as well.
+    attempts_remaining = @submitButton.data('attempts-remaining')
```

Comment:
Can you rewrite this as below? I believe the logic is the same, and I find it easier to understand.

If I've screwed up the logic, ignore this suggestion!

````
    if enable
      if attempts_remaining != 0
        @submitButton.removeAttr 'disabled'
        # the is-disabled is needed for Lettuce test below
        # lms/djangoapps/courseware/features/problems.py submit_problem(step)
        @submitButton.removeClass 'is-disabled'
      if changeText
        @submitButtonLabel.text(@submitButtonSubmitText)
    else
      @submitButton.attr({'disabled': 'disabled'})
      ....